scripts/commands/checklocalvar.lua

```diff
@@ -0,0 +1,70 @@
+---------------------------------------------------------------------------------------------------
+-- func: checklocalvar <varName> <player/mob/npc> <ID>
+-- desc: checks player or npc local variable and returns result value.
+---------------------------------------------------------------------------------------------------
+
+cmdprops =
+{
+    permission = 1,
+    parameters = "sss"
+};
+
+function error(player, msg)
+    player:PrintToPlayer(msg);
+    player:PrintToPlayer("!checklocalvar <variable name> optional <player/mob/npc> optional <ID>");
+end;
+
+function onTrigger(player, arg1, arg2, arg3)
+    local zone = player:getZoneID()
+    local varName = arg1
+    local entity = arg2
+    local targ = arg3
+
+    if (arg2 == nil) then
+        targ = player:getCursorTarget()
+        if targ == nil then
+            error(player, "no target")
+            return
+        end
+    end
+
+    if (varName == nil) then
+        error(player, "You must provide a variable name.");
+        return;
+    end
+
+    if arg2 ~= nil and arg3 == nil then
+        error(player, "no ID given for target")
+        return
+    end
```

Comment:
Might be able to make this a little more clear by bringing this up to form an `else` for the `arg2 == nil` check above:

```lua
    if arg2 == nil then
        targ = player:getCursorTarget()
        if targ == nil then
            error(player, "no target")
            return
        end
    elseif arg3 == nil then
        error(player, "no name or ID given for target")
        return
    end
```